### ISSUE TITLE
Provisioning Minishift against remote RHEL machine not possible

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#2470]
Since we are using minishift provisioner for each rhel/fedora/centos for remote provision and to check that we put following logic in the detector.go, unfortunately rhel is the id instead `redhat` for RehHat machine :( 

```golang
func (detector *MinishiftProvisionerDetector) isRHELVariantIso(osReleaseInfo *provision.OsRelease) bool {
	if osReleaseInfo.ID == "centos" || osReleaseInfo.ID == "fedora" || osReleaseInfo.ID == "redhat" {
		return true
	}
	return false
}
```

**Comment by gbraad** [comment]
This code needs to be refactored by iterating over de possible providers and ask oif they can support this ISO/platform target.

**Comment by LalatenduMohanty** [comment]
Resolved via #2472
